armory/instrument/config.py

```diff
@@ -122,345 +195,112 @@ def results(self):
-        elif adversarial:
-            if used_preds_as_labels:
-                self.wrt = "benign predictions as"
-            else:
-                self.wrt = "ground truth"
-            self.task_type = "adversarial"
-        else:
-            self.wrt = "ground truth"
-            self.task_type = "benign"
-
-    def _write(self, name, batch, result):
-        # TODO: once metrics have also been updated, rewrite to be less error prone
-        #    E.g., if someone renames this from "benign_word_error_rate" to "benign_wer"
-        if "word_error_rate" in name:
-            if "total_word_error_rate" not in name:
-                result = metrics.get("total_wer")(result)
-            total, (num, denom) = result
-            f_result = f"total={total:.2%}, {num}/{denom}"
-        elif "hota_metrics" in name:
-            mean_results = {k: v for k, v in result.items() if "mean" in k}
-            f_result = f"{mean_results}"
-        elif "entailment" in name:
-            if "total_entailment" not in name:
-                result = metrics.get("total_entailment")(result)
-            total = sum(result.values())
-            f_result = (
-                f"contradiction: {result['contradiction']}/{total}, "
-                f"neutral: {result['neutral']}/{total}, "
-                f"entailment: {result['entailment']}/{total}"
-            )
-        elif any(m in name for m in MEAN_AP_METRICS):
-            if "input_to" in name:
-                for m in MEAN_AP_METRICS:
-                    if m in name:
-                        metric = metrics.get(m)
-                        result = metrics.task.MeanAP(metric)(result)
-                        break
-            f_result = f"{result}"
-        elif any(m in name for m in QUANTITY_METRICS):
-            # Don't include % symbol
-            f_result = f"{np.mean(result):.2}"
-        elif "object_detection_mAP_tide" in name:
-            f_result = result
-        else:
-            f_result = f"{np.mean(result):.2%}"
-        log.success(
-            f"{name} on {self.task_type} examples w.r.t. {self.wrt} labels: {f_result}"
+        meter = Meter(
+            f"perturbation_{name}",
+            metric,
+            "scenario.x",
+            "scenario.x_adv",
+            final=final,
+            final_name=f"perturbation_mean_{name}",
+            record_final_only=record_final_only,
         )
+        meters.append(meter)
+    return meters
 
 
-def _task_metric(
+def task_meter(
     name,
+    prefix,
     metric_kwargs,
+    y,
+    y_pred,
     use_mean=True,
-    include_benign=True,
-    include_adversarial=True,
-    include_targeted=True,
     record_final_only=True,
+    suffix="",
 ):
     """
-    Return list of meters generated for this specific task
+    Return meter generated for this specific task
     """
-    meters = []
     metric = metrics.get(name)
+    result_formatter = metrics.get_result_formatter(name)
```

Comment:
Minor comment, not sure what best practices are here, but would suggest moving this closer to `return Meter` since `result_formatter` is only used there

Reply:
I should have actually used in `GlobalMeter` too, fixing.